File: week12/week12-3.cpp
///week12-3.cpp 列出 n以下的全部質數
#include <stdio.h>
int isPrime(int n)
{///質數不能整除
    for(int i=2;i<n;i++){
        if(n%i==0) return 0;///失敗
    }
    return 1;///成功!!!結尾不能有功，有毒!!
}

int main()
{
    printf("請輸入1個數");
    int n;
    scanf("%d",&n);
    int ans=0;
    for(int i=2;i<=n;i++){///列出
        if(isPrime(i)){
            printf("%d\n",i);
            ans++;
        }
    }
    printf("總共有%d個質數\n",ans);
}

File: week12/week12-3.py
#week12-3.py 質數的篩子法
table = [True]*240000
ans = 0 #有幾個質數
for i in range(2, 240000):
  if table[i]==True: #找到了，他是質數，因為沒有被殺死，當國王
    ans +=1
    for k in range(i*i, 240000, i): table[k] = False #國王繼位後，把他的兄弟殺光光
print('找到',ans,'個質數')